core/vm.cpp

```diff
@@ -3406,4 +3408,14 @@ std::vector<std::string> jsonnet_vm_execute_stream(Allocator *alloc, const AST *
     return vm.manifestStream(string_output);
 }
 
+inline int64_t safeDoubleToInt64(double value, const internal::LocationRange& loc) {
+    if (std::isnan(value) || std::isinf(value)) {
+        throw internal::StaticError(loc, "numeric value is not finite");
+    }
+    if (value < static_cast<double>(INT64_MIN) || value > static_cast<double>(INT64_MAX)) {
```

Comment:
If going for truly safe bitwise operations, range should be $-(2^{53} – 1)$ to $2^{53} – 1$ (Because f64 mantissa is only 53 bits).

Reply:
Fair point and I think that's the likely intent of the spec, while it doesn't explicitly mention range limitations or precision concerns. I can change that, thanks!

Reply:
Pushed and updated the PR description.